markets/bfp-market/contracts/modules/LiquidationModule.sol

```diff
@@ -314,42 +314,22 @@ contract LiquidationModule is ILiquidationModule {
         );
     }
 
-    /// @dev Returns the reward for liquidating margin.
-    function getMarginLiquidationOnlyReward(
-        uint256 collateralValue,
-        PerpMarketConfiguration.Data storage marketConfig,
-        PerpMarketConfiguration.GlobalData storage globalConfig
-    ) internal view returns (uint256) {
-        uint256 ethPrice = globalConfig
-            .oracleManager
-            .process(globalConfig.ethOracleNodeId)
-            .price
-            .toUint();
-        uint256 liqExecutionCostInUsd = ethPrice.mulDecimal(
-            block.basefee * globalConfig.keeperLiquidateMarginGasUnits
-        );
-
-        uint256 liqFeeInUsd = MathUtil.max(
-            liqExecutionCostInUsd.mulDecimal(
-                DecimalMath.UNIT + globalConfig.keeperProfitMarginPercent
-            ),
-            liqExecutionCostInUsd + globalConfig.keeperProfitMarginUsd
-        );
-        uint256 liqFeeWithRewardInUsd = liqFeeInUsd +
-            collateralValue.mulDecimal(marketConfig.liquidationRewardPercent);
-
-        return MathUtil.min(liqFeeWithRewardInUsd, globalConfig.maxKeeperFeeUsd);
-    }
-
     /// @inheritdoc ILiquidationModule
     function liquidateMarginOnly(uint128 accountId, uint128 marketId) external {
         FeatureFlag.ensureAccessToFeature(Flags.LIQUIDATE_MARGIN_ONLY);
 
         Account.exists(accountId);
         PerpMarket.Data storage market = PerpMarket.exists(marketId);
         uint256 oraclePrice = market.getOraclePrice();
-
-        if (!Margin.isMarginLiquidatable(accountId, market, oraclePrice)) {
+        Margin.MarginValues memory marginValues = Margin.getMarginUsd(
+            accountId,
+            market,
+            oraclePrice
+        );
+        if (
+            !Margin.isMarginLiquidatable(accountId, market, marginValues) ||
+            marginValues.collateralUsd == 0
```

Comment:
Minor: You could probably put `marginValues.collateralUsd == 0` before the `!Margin.isMarginLiquidatable(accountId, market, marginValues)` to be a little bit more efficient.